### ISSUE TITLE
Trying to keep up with project!

### ISSUE THREAD

**Issue by DJBHollis** [dita-ot/dita-ot#2388]
I'm finding it very difficult and frustrating to keep up with this project. I'm using GitHub desktop, and I think it is tripping up over the sub projects.

I would appreciate an explanation about how to keep up with this project, please. What the sub projects are, and how to keep up with them. Whether to sync/pull sub projects first, then the main project?

Some error messages are attached.

I am in contact with GitHub Support, but I think an explanation here would be useful, please. Mainly for folks who are not adept with GitHub.

i have found two sub projects: 
dita-ot/docs
dita-ot/testsuite

I cloned these to directories within the main area, but it still trips over docs. This might be because I cloned `docs` to `docsrc`.

It also trips up over `src/main/plugins/org.dita-ot.html/`.

It's not the first time I've been here, and previously I just trashed everything and started over. I'm sure there's a better way!

The commit I'm expected to make is similar to:
```
src/main/docsrc
- Subproject commit 4fd9a8263cf1944f307bab3b5e59c02772905e5c
+ Subproject commit aade874897f8adf0fd9b85b3a72e3daba403c02a
```

The error message if I ignore this is:
```
Updating 8ff4fa1..8f0f81a

error: Your local changes to the following files would be overwritten by merge:
src/test/testsuite
Please, commit your changes or stash them before you can merge.
Aborting
(1)
```

GitHub Support suggested I did this:
```
git status
git log -5
```

Which gives:
```
On branch develop
Your branch is behind 'origin/develop' by 11 commits, and can be fast-forwarded.
 (use "git pull" to update your local branch)

Changes to be committed:
 (use "git reset HEAD <file>..." to unstage)

	deleted:    src/main/docsrc
	deleted:    src/test/testsuite

Untracked files:
 (use "git add <file>..." to include in what will be committed)

	src/main/docsrc/
	src/main/plugins/org.dita-ot.html/
	src/test/testsuite/


git log -5

commit 8ff4fa11a6fc7958ae297848260fc6abaf560ade
Author: Jarno Elovirta <jarno@elovirta.com>
Date:   Sun May 22 13:20:45 2016 +0300

   Extract footnote callout generation to callout mode in PDF

   Use variables to reduce code duplication in callout and footnote ID generation.

commit 7453bf05fb2a4817612eafa32055dc57de1cf422
Merge: 9fb8a5e 11c4c68
Author: Jarno Elovirta <jelovirt@gmail.com>
Date:   Sun May 22 12:56:02 2016 +0300

   Merge pull request #2364 from robander/hotfix/2359

   Add linking for footnotes #2359

commit 9fb8a5e5e2d89451f7af99842ed27f23f0b02982
Merge: 2bb957f 870ffa6
Author: Jarno Elovirta <jelovirt@gmail.com>
Date:   Sun May 22 11:47:05 2016 +0300

   Merge pull request #2369 from infotexture/feature/sass-comments

   Use Sass `// comment` syntax per #2218

commit 2bb957f0ff44f973cff678a74f0f981d2ec5bcbf
Author: Jarno Elovirta <jarno@elovirta.com>
Date:   Sun May 22 10:04:55 2016 +0300

   Fix key scope reader with scope in root element of root map #2339

commit 4fe1a590dff4ba96690ce07be21299a03af3478d
Author: Jarno Elovirta <jarno@elovirta.com>
Date:   Sat May 21 21:07:57 2016 +0300

   Add deployment to Travis configuration
```

**Comment by xephon2** [comment]
The [REAMDE.md](https://github.com/dita-ot/dita-ot) explains how the DITA-OT should be built. All submodules, like the docs, are pulled automatically.

**Comment by robander** [comment]
As @xephon2 said, the [README ](https://github.com/dita-ot/dita-ot) markdown file lists the basic instructions to build. The contents of the README are displayed directly on that page if you scroll down past the file lists. As of today, it should just be four steps.

**Comment by DJBHollis** [comment]
Thanks, but it's not to do with building the project, but how GitHub works.

For instance, a git pull from GitHub desktop produces an error:

fatal: no submodule mapping found in .gitmodules for path 'src/main/plugins/org.dita-ot.html'
(128)

A git pull from the command line produces:

Already up-to-date.

So there seems to be a discrepancy between GitHub desktop and the command line? I'm discussing this with support.

Why is this important? For a couple of reasons:

The templated project home page has a nice big green button: "Clone or download".

This is how I cloned the project to start with.

The fact that there's a convenient GUI based desktop application led me to believe that contributing to the project would be nice and easy. I assumed it would work much the same as SVN, and I didn't even realise that Git has command line tools.

The point is that I'm not a developer, although I have hacked lots of different types of software, with all sorts of tools, at different times over the years. I've just never used Git until now.

The issues with Git software has nothing to do with this project, except that there might be other folk like me who would like to contribute, and are wondering how to do so.

Yes, there is the file CONTRIBUTING.md, which has a nice all uppercase filename. But it can't compete with a nice big, shiny green button!

What, if anything, can folk do about this? I don't know. I'm simply sharing my experiences, that's all.

Who knows? it might help someone else who isn't a developer, but wants to contribute, somehow.

**Comment by keberlein** [comment]
Git is not SVN. It is very powerful, and accordingly not completely simple to use. I worked through the course on Git that is available through Lynda.com. You can get free access to the course for 10 days. Good luck!

**Comment by georgebina** [comment]
Hi David, 

I was wondering, on what parts of the project you would like to contribute: Java, XSLT, documentation, something else?

For the documentation part we hope to make that easier with the oXygen XML Web Author which allows you to access the documentation directly from GitHub and when you try to commit it will 

- fork the repository automatically into your account
- create a new branch from the repository edited branch
- commit your changes on that branch
- send a pull request from that branch against the repository edited branch
- switch the edited document to the newly created branch from your fork so any other changes that you make further will be part of the already started pull request

Here it is a link that opens the root map for the DITA-OT documentation from the **develop** branch: 
[https://www.oxygenxml.com/webapp-demo-aws/app/oxygen.html?url=github%3A%2F%2FgetFileContent%2Fdita-ot%2Fdocs%2Fdevelop%2Fuserguide-book.ditamap](https://www.oxygenxml.com/webapp-demo-aws/app/oxygen.html?url=github%3A%2F%2FgetFileContent%2Fdita-ot%2Fdocs%2Fdevelop%2Fuserguide-book.ditamap)

If you want to contribute Java code or XSLT scripts then I guess you will have to get more familiar with Git and GitHub and maybe use a desktop client like, I find SourceTree very useful.

Regards,
George

**Comment by DJBHollis** [comment]
Thanks for all the comments.

> Git is not SVN.

Very true! But how many doc teams use SVN as a basic VCS? The GitHub desktop client also looks simple.

I plan to contribute some XSLT PDF2 hacks. I do like the oXygen web app link to the documentation, though.

**Comment by DJBHollis** [comment]
There is no mention of `https://github.com/dita-ot/org.dita-ot.html` in `.gitmodules`.

Should there be?

**Comment by jelovirt** [comment]
No.

**Comment by xephon2** [comment]
@DJBHollis: The mentioned plugin is being used to build the website of the DITA-OT. This plugin is not and should not be part of the DITA-OT core package. Unfortunately the description is very short and may be confusing:

> DITA-OT site HTML plug-in

A few months ago I also stumbled over this plugin and asked the same question to myself. I added an issue to the plugin and recommended to explain the intended usage in a README.md [#3](https://github.com/dita-ot/org.dita-ot.html/issues/3).
